src/views/transaction-details/transaction-details.view.tsx

```diff
@@ -12,20 +13,66 @@ import Chain from "src/views/transaction-details/components/chain/chain";
 import { useBridgeContext } from "src/contexts/bridge.context";
 import { useProvidersContext } from "src/contexts/providers.context";
 import { useUIContext } from "src/contexts/ui.context";
+import { useEnvContext } from "src/contexts/env.context";
 import { parseError } from "src/adapters/error";
 import { AsyncTask, isMetamaskUserRejectedRequestError } from "src/utils/types";
-import { getTransactionStatus } from "src/utils/labels";
-import { Transaction } from "src/domain";
+import { getTransactionStatus, getChainName } from "src/utils/labels";
 import { formatTokenAmount } from "src/utils/amounts";
+import { Transaction, Token } from "src/domain";
+
+interface HistoricalFees {
+  step1?: string;
+  step2?: string;
+}
+
+const calculateTxFee = (txResponse: TransactionResponse, token: Token): string | undefined => {
+  if (txResponse.maxFeePerGas && txResponse.maxPriorityFeePerGas) {
```

Comment:
We already have this logic in the `estimateGasPrice` of the `BridgeContext`. Can we centralize this in a util that checks if the `FeeData` is from `EIP-1559` or not and just returns the fee?

Reply:
Done